spacy_llm/pipeline/llm.py

```diff
@@ -126,6 +134,9 @@ def pipe(self, stream: Iterable[Doc], *, batch_size: int = 128) -> Iterator[Doc]
         batch_size (int): The number of documents to buffer.
         YIELDS (Doc): Processed documents in order.
         """
+        if not Doc.has_extension("llm_io"):
```

Comment:
This needs to be added to `__call__` as well. Since our backend + response parser always work on batches I think putting this in a util method would work best. 

We're deferring this logging PR (#80) for 0.3.0 but the `_process_docs` method could be added to this PR I think. Then the extension installation (+ setting of the prompt/response) only needs to happen in one spot.

[_process_docs method](https://github.com/explosion/spacy-llm/blob/3e7c481a0300f3fd227df98b19785373305c3c81/spacy_llm/pipeline/llm.py#L242)

Reply:
Ah absolutely, thanks! I'll wait for your PR to be merged then.

Reply:
I agree with Kabir, but I don't think we need to wait for https://github.com/explosion/spacy-llm/pull/80 to be merged. I'd recommend copying over `_process_docs()` and adding setting the extension.

Reply:
Oh yeah, that's what I meant, I think this PR should be merged first but we should copy over the core of `_process_docs` so we only need to install the extensions in 1 method.

Reply:
Alright!

Reply:
Done, thanks.